discovery-handlers/onvif/src/discovery_impl.rs

```diff
@@ -341,7 +345,7 @@ pub mod util {
             "simple_onvif_discover - uris discovered by udp broadcast {:?}",
             broadcast_responses
         );
-        let mut filtered_uris = Vec::new();
+        let mut filtered_uris = std::collections::HashSet::new();
         broadcast_responses.into_iter().for_each(|r| {
             filtered_uris.extend(get_scope_filtered_uris_from_discovery_response(
```

Comment:
this seem like a fine solution.  would it make sense to pass filtered_uris into get_scope_filtered_uris_from_discovery_response so that you only execute_filter once for duplicate responses?

Reply:
`filtered_uris` are created by `get_scope_filtered_uris_from_discovery_response` which gets the uris from each discovery response.